packages/test/mocha-test-setup/src/mochaHooks.ts

```diff
@@ -24,112 +24,119 @@ const propsDict =
 		: undefined;
 
 const _global: any = global;
-class TestLogger implements ITelemetryBufferedLogger {
+
+/**
+ * A logger that adds context about the current test run to all events logged through it, like the test variant being
+ * run (odsp, r11s, etc) and the name of the current test (while in the context of a particular test running).
+ */
+class FluidTestRunLogger implements ITelemetryBufferedLogger {
+	private currentTestName: string | undefined;
+
 	send(event: ITelemetryBaseEvent) {
 		// TODO: Remove when issue #7061 is resolved.
 		// Don't log this event as we generate too much.
 		if (event.eventName === "fluid:telemetry:RouterliciousDriver:readBlob_end") {
 			return;
 		}
 
-		// The test logger is currently instantiated once and for each event triggered between begin and
-		// end of a test, in case the testName is undefined, we will use the currentTestName.
-		event.testName = this.testName ?? currentTestName;
+		if (this.currentTestName !== undefined) {
+			event.testName = this.currentTestName;
+		}
 		event.testVariant = testVariant;
-		event.hostName = pkgName;
 		this.parentLogger.send({
 			...event,
-			// If there's an override for the hostName in FLUID_LOGGER_PROPS,
-			// display that in the telemetry instead of the package name.
+			// Setting hostname to pkgName is the behavior we had for a long time, so keeping it just in case.
+			// But prefer a value set through FLUID_LOGGER_PROPS if it exists.
 			hostName: propsDict?.hostName ?? pkgName,
 			details: JSON.stringify(propsDict),
 		});
 	}
 	async flush() {
 		return this.parentLogger.flush();
 	}
-	constructor(
-		private readonly parentLogger: ITelemetryBufferedLogger,
-		private readonly testName: string | undefined,
-	) {}
+	constructor(private readonly parentLogger: ITelemetryBufferedLogger) {}
+
+	/**
+	 * Sets the test that is currently running.
+	 * The test name will be included in all events logged through the logger until {@link clearCurentTest} is called.
+	 * @param testName - The name of the test that is currently running.
+	 */
+	public setCurrentTest(testName: string) {
```

Comment:
Methods to update this bit of inner state (the name of the current test) in the logger class seem clearer than it using an external variable `currentTestName` which other code could update at any time.